infrastructure/src/main/java/org/corfudb/infrastructure/NettyServerRouter.java

```diff
@@ -90,25 +91,13 @@ public void sendResponse(ChannelHandlerContext ctx, CorfuMsg inMsg, CorfuMsg out
         log.trace("Sent response: {}", outMsg);
     }
 
-    /**
-     * Validate the epoch of a CorfuMsg, and send a WRONG_EPOCH response if
-     * the server is in the wrong epoch. Ignored if the message type is reset (which
-     * is valid in any epoch).
-     *
-     * @param msg The incoming message to validate.
-     * @param ctx The context of the channel handler.
-     * @return True, if the epoch is correct, but false otherwise.
-     */
-    public boolean validateEpoch(CorfuMsg msg, ChannelHandlerContext ctx) {
-        long serverEpoch = getServerEpoch();
-        if (!msg.getMsgType().ignoreEpoch && msg.getEpoch() != serverEpoch) {
-            sendResponse(ctx, msg, new CorfuPayloadMsg<>(CorfuMsgType.WRONG_EPOCH,
-                    serverEpoch));
-            log.trace("Incoming message with wrong epoch, got {}, expected {}, message was: {}",
-                    msg.getEpoch(), serverEpoch, msg);
-            return false;
+    @Override
+    public Optional<Layout> getCurrentLayout() {
+        if(!serverMap.containsKey(LayoutServer.class)){
+            throw new IllegalStateException("Layout server is not in a server map.");
         }
-        return true;
+        LayoutServer layoutServer = (LayoutServer) serverMap.get(LayoutServer.class);
```

Comment:
The router shouldn't depend on the `LayoutServer` , its cleaner to just use `serverContext`.

Reply:
That was the first choice that I tried but I have encountered a couple of issues. 
This will create a circular dependency between a context and a router, and force one more component to depend on it, making it even more of a god object. 
This will also require more changes to be done to the TestServerRouter and the TestServer, which are messy to implement. 
Since the LayoutServer is the owner of the layout, and the router has access to the servers, I think that it's viable to retrieve a layout from a server.

Reply:
Edit: Although there is a circular dependency, this approach seems to be more consistent with the testing we are doing for the single servers and removes the incorrect approach of ignoring the routing validation.